source/common/runtime/runtime_impl.cc

```diff
@@ -208,10 +208,16 @@ void SnapshotImpl::walkDirectory(const std::string& path, const std::string& pre
       full_prefix = prefix + "." + entry->d_name;
     }
 
-    if (entry->d_type == DT_DIR && std::string(entry->d_name) != "." &&
+    struct stat stat_result;
+    int rc = ::stat(full_path.c_str(), &stat_result);
+    if (rc != 0) {
+      throw EnvoyException(fmt::format("unable to stat file: '{}'", full_path));
```

Comment:
Can we add a test for this case? Perhaps move to utility function so can test with a bogus path?

Reply:
I can add some test coverage if I move stat() to OsSysCalls.  

I think in the real world, the only way this fails is if someone deletes/moves the file after readdir() has returned it but before we stat() it.

Reply:
Either that or make it RELEASE_ASSERT. Up to you.

Reply:
(I test hack would be to just move the stat call to a small helper, and then pass in a bad file name, and I think it should fail and throw).